### PR TITLE
B3 login pages

### PR DESCRIPTION
**Pull request by orangejenny**
@sravfeyn / @biyeun 

QA ticket: http://manage.dimagi.com/default.asp?221766

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 34 errors:

* corehq/apps/domain/forms.py, line 996 - E231 missing whitespace after ':'
* corehq/apps/domain/urls.py, line 84 - E231 missing whitespace after ':'
* corehq/apps/domain/urls.py, line 89 - E231 missing whitespace after ':'
* corehq/apps/domain/urls.py, line 89 - E501 line too long (130 > 115 characters)
* corehq/apps/domain/urls.py, line 90 - E231 missing whitespace after ':'
* corehq/apps/domain/urls.py, line 90 - E501 line too long (155 > 115 characters)
* corehq/apps/domain/urls.py, line 93 - E231 missing whitespace after ':'
* corehq/apps/domain/urls.py, line 93 - E501 line too long (134 > 115 characters)
* corehq/apps/domain/urls.py, line 98 - E127 continuation line over-indented for visual indent
* corehq/apps/domain/urls.py, line 99 - E231 missing whitespace after ':'
* corehq/apps/domain/urls.py, line 99 - E501 line too long (131 > 115 characters)
* corehq/apps/domain/urls.py, line 103 - E231 missing whitespace after ':'
* corehq/apps/domain/views.py, line 147 - E231 missing whitespace after ':'
* corehq/apps/hqwebapp/forms.py, line 24 - E231 missing whitespace after ':'
* corehq/apps/hqwebapp/forms.py, line 25 - E231 missing whitespace after ':'
* corehq/apps/hqwebapp/forms.py, line 53 - E127 continuation line over-indented for visual indent
* corehq/apps/hqwebapp/forms.py, line 53 - E231 missing whitespace after ':'
* corehq/apps/registration/forms.py, line 34 - E127 continuation line over-indented for visual indent
* corehq/apps/registration/forms.py, line 34 - E231 missing whitespace after ':'
* corehq/apps/registration/forms.py, line 67 - E127 continuation line over-indented for visual indent
* corehq/apps/registration/forms.py, line 68 - E231 missing whitespace after ':'
* corehq/apps/registration/forms.py, line 72 - E231 missing whitespace after ':'
* corehq/apps/registration/forms.py, line 77 - E121 continuation line under-indented for hanging indent
* corehq/apps/registration/forms.py, line 77 - E231 missing whitespace after ':'
* corehq/apps/registration/forms.py, line 78 - E231 missing whitespace after ':'
* corehq/apps/registration/views.py, line 81 - E231 missing whitespace after ':'
* corehq/apps/registration/views.py, line 92 - E231 missing whitespace after ':'
* corehq/apps/registration/views.py, line 113 - E231 missing whitespace after ':'
* corehq/apps/registration/views.py, line 156 - E231 missing whitespace after ':'
* corehq/apps/registration/views.py, line 216 - E231 missing whitespace after ':'
* corehq/apps/registration/views.py, line 223 - E231 missing whitespace after ':'
* corehq/apps/registration/views.py, line 247 - E231 missing whitespace after ':'
* corehq/apps/users/views/__init__.py, line 650 - E231 missing whitespace after ':'
* corehq/apps/users/views/__init__.py, line 652 - E231 missing whitespace after ':'

**Comment by dimagimon** [comment]
There are 8 errors:

* corehq/apps/domain/urls.py, line 89 - E501 line too long (131 > 115 characters)
* corehq/apps/domain/urls.py, line 90 - E501 line too long (158 > 115 characters)
* corehq/apps/domain/urls.py, line 93 - E501 line too long (137 > 115 characters)
* corehq/apps/domain/urls.py, line 98 - E127 continuation line over-indented for visual indent
* corehq/apps/domain/urls.py, line 99 - E501 line too long (134 > 115 characters)
* corehq/apps/registration/forms.py, line 34 - E127 continuation line over-indented for visual indent
* corehq/apps/registration/forms.py, line 67 - E127 continuation line over-indented for visual indent
* corehq/apps/registration/forms.py, line 77 - E121 continuation line under-indented for hanging indent

File: corehq/apps/domain/urls.py [hunk 1]
```diff
@@ -80,20 +79,31 @@ def extend(d1, d2):
...
+        url(r'^accounts/password_reset_email/done/$', 'password_reset_done',
+            extend(auth_pages_path('password_reset_done.html'),
+                   {'extra_context': {'current_page': {'page_name': 'Reset My Password'}}}),
             name='password_reset_done'),
 
         url(r'^accounts/password_reset_confirm/(?P<uidb64>[0-9A-Za-z_\-]+)/(?P<token>.+)/$',
             PasswordResetView.as_view(),  extend(auth_pages_path('password_reset_confirm.html'),
-                                                    {'set_password_form': HQSetPasswordForm}), name=PasswordResetView.urlname),
-        url(r'^accounts/password_reset_confirm/done/$', 'password_reset_complete', auth_pages_path('password_reset_complete.html'),
+                                                    {'set_password_form': HQSetPasswordForm,
```
Comment (dimagimon) [inline corehq/apps/domain/urls.py]: E127 continuation line over-indented for visual indent

File: corehq/apps/registration/forms.py [hunk 2]
```diff
@@ -30,7 +30,8 @@ class DomainRegistrationForm(forms.Form):
     max_name_length = 25
 
     org = forms.CharField(widget=forms.HiddenInput(), required=False)
-    hr_name = forms.CharField(label=_('Project Name'), max_length=max_name_length)
+    hr_name = forms.CharField(label=_('Project Name'), max_length=max_name_length,
+                                      widget=forms.TextInput(attrs={'class': 'form-control'}))
```
Comment (dimagimon) [inline corehq/apps/registration/forms.py]: E127 continuation line over-indented for visual indent

File: corehq/apps/registration/forms.py [hunk 3]
```diff
@@ -63,14 +64,19 @@ class NewWebUserRegistrationForm(NoAutocompleteMixin, DomainRegistrationForm):
     """
     full_name = forms.CharField(label=_('Full Name'),
                                 max_length=User._meta.get_field('first_name').max_length +
-                                           User._meta.get_field('last_name').max_length + 1)
+                                           User._meta.get_field('last_name').max_length + 1,
```
Comment (dimagimon) [inline corehq/apps/registration/forms.py]: E127 continuation line over-indented for visual indent

File: corehq/apps/registration/forms.py [hunk 4]
```diff
@@ -63,14 +64,19 @@ class NewWebUserRegistrationForm(NoAutocompleteMixin, DomainRegistrationForm):
...
                              max_length=User._meta.get_field('email').max_length,
-                             help_text=_('You will use this email to log in.'))
+                             help_text=_('You will use this email to log in.'),
+                             widget=forms.TextInput(attrs={'class': 'form-control'}))
     password = forms.CharField(label=_('Create Password'),
                                max_length=max_pwd,
                                widget=forms.PasswordInput(render_value=False,
-                                                          attrs={'data-bind': "value: password, valueUpdate: 'input'"}),
+                                                          attrs={
+                                                            'data-bind': "value: password, valueUpdate: 'input'",
```
Comment (dimagimon) [inline corehq/apps/registration/forms.py]: E121 continuation line under-indented for hanging indent

File: corehq/apps/domain/urls.py [hunk 5]
```diff
@@ -80,20 +79,31 @@ def extend(d1, d2):
     ) +\
     patterns('django.contrib.auth.views',
         url(r'^accounts/password_change/$', 'password_change', auth_pages_path('password_change_form.html'), name='password_change'),
-        url(r'^accounts/password_change_done/$', 'password_change_done', auth_pages_path('password_change_done.html'),
+        url(r'^accounts/password_change_done/$', 'password_change_done',
+            extend(auth_pages_path('password_change_done.html'),
+                   {'extra_context': {'current_page': {'page_name': 'Password Change Complete'}}}),
```
Comment (orangejenny) [inline corehq/apps/domain/urls.py]: all of these `extra_context` changes are hackish, i admit

File: corehq/apps/domain/views.py [hunk 6]
```diff
@@ -170,20 +171,6 @@ def select(request, domain_select_template='domain/select.html', do_not_redirect
         return render(request, domain_select_template, additional_context)
 
 
-@require_superuser
```
Comment (orangejenny) [inline corehq/apps/domain/views.py]: @amsagoff approved killing this page

**Comment by biyeun** [comment]
@orangejenny conflicts

@dimagi/product weren't we going to completely redo this view? jenny, is this a simple switch? I had put these off because of the redo requirements...surprised that wasn't communicated by product.

**Comment by amsagoff** [comment]
Hey,

Is this about the signup experience redo? I wasn't sure when you would have
time to actually work on that, and didn't want it to hold up the B3
migration.

**Comment by nickpell** [comment]
merged master

**Comment by biyeun** [comment]
Cool, no worries. Thanks @orangejenny :tada:

File: corehq/apps/domain/urls.py [hunk 7]
```diff
@@ -53,9 +54,12 @@ def exception_safe_password_reset(request, *args, **kwargs):
     try:
         return password_reset(request, *args, **kwargs)
     except None:
-        vals = {'error_msg':'There was a problem with your request',
-                'error_details':sys.exc_info(),
-                'show_homepage_link': 1 }
+        vals = {
+            'current_page': {'page_name': _('Oops!')},
+            'error_msg':'There was a problem with your request',
+            'error_details':sys.exc_info(),
```
Comment (dimagimon) [inline corehq/apps/domain/urls.py]: E231 missing whitespace after ':'

File: corehq/apps/domain/urls.py [hunk 8]
```diff
@@ -53,9 +54,12 @@ def exception_safe_password_reset(request, *args, **kwargs):
     try:
         return password_reset(request, *args, **kwargs)
     except None:
-        vals = {'error_msg':'There was a problem with your request',
-                'error_details':sys.exc_info(),
-                'show_homepage_link': 1 }
+        vals = {
+            'current_page': {'page_name': _('Oops!')},
+            'error_msg':'There was a problem with your request',
```
Comment (dimagimon) [inline corehq/apps/domain/urls.py]: E231 missing whitespace after ':'

File: corehq/apps/app_manager/templates/app_manager/partials/case_list.html [hunk 9]
```diff
@@ -203,7 +203,9 @@
                 {% trans "Select Parent First" %}
             </label>
             <div class="col-sm-4 checkbox">
-                <input type="checkbox" data-bind="checked: active"/>
+                <label>
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/case_list.html]: Didn't mean to put this change in this PR, but it's not hurting anything. It just adjusts the spacing for this checkbox.

File: corehq/apps/app_manager/templates/app_manager/partials/case_list_lookup.html [hunk 10]
```diff
@@ -11,7 +11,9 @@
             {% trans "Enable case list search callout" %}
         </label>
         <div class="col-sm-4 checkbox">
-            <input type="checkbox" data-bind="checked: lookup_enabled"/>
+            <label>
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/case_list_lookup.html]: same comment as above

**Comment by orangejenny** [comment]
QA passed, conflicts resolved, this is good to go once tests pass.

File: corehq/apps/hqwebapp/views.py [hunk 11]
```diff
@@ -66,6 +66,7 @@
 from corehq.apps.users.models import CouchUser
 from corehq.apps.users.util import format_username
 from corehq.middleware import always_allow_browser_caching
+from corehq.util.context_processors import get_per_domain_context
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/views.py]: F401 'get_per_domain_context' imported but unused

**Comment by orangejenny** [comment]
Good to go!

@biyeun / new code buddy @calellowitz

**Comment by benrudolph** [comment]
👢 strap 3️⃣  🔑

**Comment by biyeun** [comment]
YESSS :boom: 🎉

**Pull request merged**